src/features/items.ts

```diff
@@ -1,175 +1,141 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import { configureStore } from "@reduxjs/toolkit";
+import { ref } from "../FirebaseContext";
 import { ItemType } from "../types/ListItem";
 import { TagType } from "../types/Tag";
-import { linkBiscuitPrefix } from "../utils/constants";
-let itemsCounter = 0;
+import { reload } from "./loading";
+import { popSnack } from "./snack";
 
 export const LOAD_ITEMS = "LOAD_ITEMS";
 export type LoadItemsAction = {
   type: typeof LOAD_ITEMS;
+  items: ItemType[];
 };
-export const loadItems: () => LoadItemsAction = () => ({ type: LOAD_ITEMS });
+export const loadItems: (items: ItemType[]) => LoadItemsAction = (items) => ({
+  type: LOAD_ITEMS,
+  items,
+});
 
-export const DELETE_ITEM = "DELETE_ITEM";
-export type DeleteItemAction = {
-  type: typeof DELETE_ITEM;
-  listItem: ItemType;
+export const ITEM_DELETED = "ITEM_DELETED";
+export type ItemDeletedAction = {
+  type: typeof ITEM_DELETED;
+  item: ItemType;
 };
 
-export const deleteItem: (item: ItemType) => DeleteItemAction = (item) => {
+export const itemDeleted: (item: ItemType) => ItemDeletedAction = (item) => {
   return {
-    type: DELETE_ITEM,
-    listItem: item,
+    type: ITEM_DELETED,
+    item,
   };
 };
-export const ADD_NEW_ITEM = "ADD_NEW_ITEM";
-export type AddNewItemAction = {
-  type: typeof ADD_NEW_ITEM;
-  listItem: ItemType;
+
+export const ITEM_ADDED = "ITEM_ADDED";
+export type ItemAddedAction = {
+  type: typeof ITEM_ADDED;
+  item: ItemType;
 };
-export const addNewItem: (url: string) => AddNewItemAction = (url) => {
-  itemsCounter++;
+export const itemAdded: (item: ItemType) => ItemAddedAction = (item) => {
   return {
-    type: ADD_NEW_ITEM,
-    listItem: {
-      id: itemsCounter.toString(),
-      url,
-      date: new Date().getTime().toString(),
-      alreadyRead: false,
-      tags: [],
-    },
+    type: ITEM_ADDED,
+    item,
   };
 };
 
-export const ASSIGN_TAG_TO_ITEM = "ASSIGN_TAG_TO_ITEM";
-export type AssignTagToItemAction = {
-  type: typeof ASSIGN_TAG_TO_ITEM;
+export const TAG_ASSIGNED = "TAG_ASSIGNED";
+export type TagAssignedAction = {
+  type: typeof TAG_ASSIGNED;
   item: ItemType;
-  tag: TagType;
 };
-export const assignTagToItem: (
-  tag: TagType,
-  item: ItemType
-) => AssignTagToItemAction = (tag, item) => {
+export const tagAssigned: (item: ItemType) => TagAssignedAction = (item) => {
```

Comment:
Maybe a bit nitpicking but I liked the previous wording a bit better. Cause the action describes what was triggered by the user. So if I understand it properly I'd expect having here something like `assignTag` instead of `tagAssigned`.